src/topics/recent.js

```diff
@@ -63,6 +64,10 @@ module.exports = function(Topics) {
 
 	Topics.updateRecent = function(tid, timestamp, callback) {
 		callback = callback || function() {};
-		db.sortedSetAdd('topics:recent', timestamp, tid, callback);
+		if (plugins.hasListeners('filter:topics.updateRecent')) {
+			plugins.fireHook('filter:topics.updateRecent', {tid: tid, timestamp: timestamp}, callback);
```

Comment:
Nah your current code is fine just don't change the return of the `Topics.updateRecent` function.

So the entire function would look like.

```
Topics.updateRecent = function(tid, timestamp, callback) {
		callback = callback || function() {};
		if (plugins.hasListeners('filter:topics.updateRecent')) {
			plugins.fireHook('filter:topics.updateRecent', {tid: tid, timestamp: timestamp}, function(err,data) {
				callback(err);
			});
		} else {
			db.sortedSetAdd('topics:recent', timestamp, tid, callback);
		}
	};
```